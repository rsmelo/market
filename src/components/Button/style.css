.button {
  font-family: var(--font-family);
  cursor: pointer;
  font-size: var(--font-size-default);
  text-align: center;
  border-radius: 4px;
  outline: 0;
  padding: 12px 14px;
  transition: 0.2s ease-out;
  border-style: solid;
  border-width: 1px;
  text-transform: uppercase;
}

.flat-green,
.flat-grey {
  color: var(--color-white);
}

.flat-green {
  background: var(--color-light-green);
  border-color: var(--color-light-green);
}

.flat-green:hover {
  background: var(--color-dark-green);
  border-color: var(--color-dark-green);
}

.flat-grey {
  background: var(--color-ligther-grey);
  border-color: var(--color-ligther-green);
}

.flat-grey:hover {
  background-color: var(--color-ligth-grey);
  border-color: var(--color-light-ligthgreen);
}

.outline {
  background: transparent;
}

.outline-green {
  color: var(--color-light-green);
  border-color: var(--color-light-green);
}

.outline-green:hover {
  color: var(--color-dark-green);
  border-color: var(--color-dark-green);
}

.outline-grey {
  color: var(--color-ligther-grey);
  border-color: var(--color-ligther-grey);
}

.outline-grey:hover {
  color: var(--color-ligth-grey);
  border-color: var(--color-ligth-grey);
}

.button:disabled:hover,
.button:disabled {
  background: var(--color-ligther-grey);
  color: var(--color-ligth-grey);
  border-color: var(--color-ligther-grey);
  cursor: not-allowed;
}
