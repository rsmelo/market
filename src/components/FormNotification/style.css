.formNotification {
  margin-top: 1em;
  display: block;
  padding: 1em;
}

.error {
  border: 1px solid var(--color-error-ligth);
  background: var(--color-error-ligther);
}

.success {
  border-left: 1px solid var(--color-success-light);
  background: var(--color-success-lighter);
}
