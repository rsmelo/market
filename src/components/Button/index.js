import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import style from './style.css'

const Button = ({
  onClick,
  type,
  children,
  color,
  appearance,
}) => {
  const styles = classNames(
    style.button,
    style[appearance],
    style[`${appearance}-${color}`]
  )

  return (
    <button
      className={styles}
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  )
}


Button.defaultProps = {
  onClick: null,
  appearance: 'flat',
  color: 'green',
  type: 'button',
}

Button.propTypes = {
  type: PropTypes.oneOf(['submit', 'reset', 'button']),
  onClick: PropTypes.func,
  appearance: PropTypes.oneOf([
    'flat', 'outline',
  ]),
  color: PropTypes.oneOf([
    'green', 'grey',
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
}

export default Button
