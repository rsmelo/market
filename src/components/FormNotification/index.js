import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import style from './style.css'

const FormNotification = ({ children, type }) => (
  <div className={classNames(style.formNotification, style[type])}>
    {children}
  </div>
)


FormNotification.defaultProps = {
  type: 'error',
}

FormNotification.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
  type: PropTypes.oneOf(['success', 'error']),
}


export default FormNotification
