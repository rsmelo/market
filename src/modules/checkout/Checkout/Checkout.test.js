import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme'

import CheckoutWithForm, { Checkout } from './Checkout'
import Button from '../../../components/Button'
import FormNotification from '../../../components/FormNotification'

describe('Checkout', () => {
  let store
  beforeEach(() => {
    store = createStore(() => ({}))
  })
  it('renders correctly', () => {
    const props = {
      cart: {},
      handleSubmit: jest.fn(),
      onSubmit: jest.fn(),
      submitting: false,
      pristine: false,
      history: {
        push: jest.fn(),
      },
    }
    const tree = renderer
      .create(<Provider store={store}><CheckoutWithForm {...props} /></Provider>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should call onSubmit when click button submit', () => {
    const cart = {}
    const push = jest.fn()

    const onSubmit = jest.fn()
    const handleSubmit = data => onSubmit({ data, cart, push })
    const props = {
      cart: {},
      handleSubmit,
      onSubmit,
      submitting: false,
      pristine: false,
      history: {
        push,
      },
    }
    const component = mount(<Provider store={store}><CheckoutWithForm {...props} /></Provider>)
    const button = component.find(Button)
    button.simulate('click')
    expect(onSubmit).toHaveBeenCalled()
  })
  it('should show FormNotification if error is passed', () => {
    jest.mock('redux-form', () => ({ Field: 'Field' }))
    const props = {
      cart: {},
      handleSubmit: jest.fn(),
      onSubmit: jest.fn(),
      submitting: false,
      pristine: false,
      history: {
        push: jest.fn(),
      },
      error: 'teste',
    }
    const component = shallow(<Checkout {...props} />)
    const notification = component.find(FormNotification)
    expect(notification).toHaveLength(1)
  })
  it('should not show FormNotification if error is not passed', () => {
    jest.mock('redux-form', () => ({ Field: 'Field' }))
    const props = {
      cart: {},
      handleSubmit: jest.fn(),
      onSubmit: jest.fn(),
      submitting: false,
      pristine: false,
      history: {
        push: jest.fn(),
      },
    }
    const component = shallow(<Checkout {...props} />)
    const notification = component.find(FormNotification)
    expect(notification).toHaveLength(0)
  })
  it('should call handleSubmit when form submit', () => {
    jest.mock('redux-form', () => ({ Field: 'Field' }))
    const push = jest.fn()
    const onSubmit = jest.fn()
    const handleSubmit = jest.fn()
    const props = {
      cart: {},
      handleSubmit,
      onSubmit,
      submitting: false,
      pristine: false,
      history: {
        push,
      },
    }
    const component = shallow(<Checkout {...props} />)
    const form = component.find('form')
    form.simulate('submit')
    expect(handleSubmit).toHaveBeenCalled()
  })
  it('should show the message "O carrinho associado a este pedido não foi localizado" when cart is no set', () => {
    jest.mock('redux-form', () => ({ Field: 'Field' }))
    const push = jest.fn()
    const onSubmit = jest.fn()
    const handleSubmit = jest.fn()
    const props = {
      cart: null,
      handleSubmit,
      onSubmit,
      submitting: false,
      pristine: false,
      history: {
        push,
      },
    }
    const component = shallow(<Checkout {...props} />)
    const paragraph = component.find('p')
    expect(paragraph.text()).toBe('O carrinho associado a este pedido não foi localizado')
  })
})
