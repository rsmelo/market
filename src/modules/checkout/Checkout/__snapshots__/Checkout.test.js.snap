// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout renders correctly 1`] = `
<div
  className={undefined}
>
  <form
    className={undefined}
    onSubmit={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <h2
          className={undefined}
        >
          1. Dados Pessoais
        </h2>
        <label
          className={undefined}
          htmlFor="name"
        >
          Nome
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="name"
            maxLength={null}
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="email"
        >
          E-mail
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="email"
            maxLength={null}
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="personCode"
        >
          CPF
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="personCode"
            maxLength={null}
            name="personCode"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="000.000.000-00"
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="phone"
        >
          Telefone
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="phone"
            maxLength={null}
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="(00) 0000-0000"
            type="input"
            value=""
          />
        </div>
      </div>
      <div
        className={undefined}
      >
        <h2
          className={undefined}
        >
          2. Endereço
        </h2>
        <label
          className={undefined}
          htmlFor="street"
        >
          Rua
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="street"
            maxLength={null}
            name="street"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="streetNumber"
        >
          Número
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="streetNumber"
            maxLength={null}
            name="streetNumber"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="complementary"
        >
          Complemento
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="complementary"
            maxLength={null}
            name="complementary"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="neighborhood"
        >
          Bairro
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="neighborhood"
            maxLength={null}
            name="neighborhood"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="city"
        >
          Cidade
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="city"
            maxLength={null}
            name="city"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="state"
        >
          Estado
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="state"
            maxLength={null}
            name="state"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder={null}
            type="input"
            value=""
          />
        </div>
        <label
          className={undefined}
          htmlFor="zipcode"
        >
          CEP
        </label>
        <div
          className={undefined}
        >
          <input
            className={undefined}
            id="zipcode"
            maxLength={null}
            name="zipcode"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="00000-000"
            type="input"
            value=""
          />
        </div>
      </div>
      <div>
        <div
          className={undefined}
        >
          <h2
            className={undefined}
          >
            3. Pagamento
          </h2>
          <label
            className={undefined}
            htmlFor="cardNumber"
          >
            Número do cartão
          </label>
          <div
            className={undefined}
          >
            <input
              className={undefined}
              id="cardNumber"
              maxLength={null}
              name="cardNumber"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="0000 0000 0000 0000"
              type="input"
              value=""
            />
          </div>
          <label
            className={undefined}
            htmlFor="cardHolderName"
          >
            Nome do titular
          </label>
          <div
            className={undefined}
          >
            <input
              className={undefined}
              id="cardHolderName"
              maxLength={null}
              name="cardHolderName"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder={null}
              type="input"
              value=""
            />
          </div>
          <label
            className={undefined}
            htmlFor="cardExpirationDate"
          >
            Expira em
          </label>
          <div
            className={undefined}
          >
            <input
              className={undefined}
              id="cardExpirationDate"
              maxLength={null}
              name="cardExpirationDate"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="00/00"
              type="input"
              value=""
            />
          </div>
          <label
            className={undefined}
            htmlFor="cardCvv"
          >
            Código de segurança
          </label>
          <div
            className={undefined}
          >
            <input
              className={undefined}
              id="cardCvv"
              maxLength={4}
              name="cardCvv"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder={null}
              type="input"
              value=""
            />
          </div>
        </div>
        <div
          className={undefined}
        >
          <button
            className=""
            disabled={true}
            onClick={null}
            type="submit"
          >
            Finalizar compra
          </button>
        </div>
      </div>
    </div>
    <div />
  </form>
</div>
`;
